### PR TITLE
chore: remove name tag as export

### PR DESCRIPTION
**Pull request by chrisdholt**
# Description
I noticed that we were exporting the name tag example, which I don't think we want to actually ship. Considering our goal with fast-components is to export "all the things" that help us compose the components, I thought we should export the form-associated abstract class.

## Issue type checklist

- [x] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

**Comment by janechu** [comment]
@nicholasrice @chrisdholt should we be exporting the form-associated logic, or was there some reason it was not being exported?

**Comment by chrisdholt** [comment]
> @nicholasrice @chrisdholt should we be exporting the form-associated logic, or was there some reason it was not being exported?

I think we should export - I mentioned this in the description, but I believe the goal is to enable folks to compose their own stuff as well. I don't see harm in exporting this, but if there was reason to not do so I can close this.

**Comment by nicholasrice** [comment]
It is likely to change pretty significantly and maybe even go away so I wanted to keep it internal until our API stabilized.

File: packages/fast-components/src/index.ts [hunk 1]
```diff
@@ -1,2 +1,2 @@
-export * from "./name-tag";
 export * from "./design-system-provider";
+export * from "./form-associated";
```
Comment (nicholasrice) [inline packages/fast-components/src/index.ts]: I don't think we should export this - at least yet - for the reasons I outlined previously.

Comment (chrisdholt) [reply packages/fast-components/src/index.ts]: Pulled this export based on our convo

**Pull request merged**